/**
 * e8yes demo web server.
 *
 * <p>Copyright (C) 2020 Chifeng Wen {daviesx66@gmail.com}
 *
 * <p>This program is free software: you can redistribute it and/or modify it under the terms of the
 * GNU General Public License as published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * <p>You should have received a copy of the GNU General Public License along with this program. If
 * not, see <http://www.gnu.org/licenses/>.
 */

syntax = "proto3";

package e8;

enum NodeType {
    NDT_INVALID = 0;
    NDT_FILE_STORE = 1;
    NDT_MESSAGE_STORE = 2;
    NDT_TASK_EXECUTOR = 3;
}

message NodeState {
    int64 clock = 1;
    bytes ip_address = 2;
    string name = 3;
    repeated NodeType types = 4;
    repeated int32 ports = 5;
}
